integration/testnet/network.go

```diff
@@ -752,47 +718,41 @@ func (net *FlowNetwork) AddObserver(t *testing.T, ctx context.Context, conf *Obs
 			"--profiler-dir=/profiler",
 			"--profiler-interval=2m",
 		},
-
-		ExposedPorts: nat.PortSet{
-			"9000": struct{}{},
-			"9001": struct{}{},
-			"8000": struct{}{},
-		},
 	}
 	containerHostConfig := &container.HostConfig{
 		Binds: []string{
 			fmt.Sprintf("%s:%s:rw", flowDataDir, "/data"),
 			fmt.Sprintf("%s:%s:rw", flowProfilerDir, "/profiler"),
 			fmt.Sprintf("%s:%s:ro", nodeBootstrapDir, "/bootstrap"),
 		},
-		PortBindings: nat.PortMap{
-			"9000": []nat.PortBinding{{HostIP: "0.0.0.0", HostPort: observerUnsecurePort}},
-			"9001": []nat.PortBinding{{HostIP: "0.0.0.0", HostPort: observerSecurePort}},
-			"8000": []nat.PortBinding{{HostIP: "0.0.0.0", HostPort: observerHttpPort}},
-		},
 	}
 
 	containerOpts := testingdock.ContainerOpts{
-		ForcePull:   false,
-		Config:      containerConfig,
-		HostConfig:  containerHostConfig,
-		Name:        conf.ObserverName,
-		HealthCheck: testingdock.HealthCheckCustom(healthcheckAccessGRPC(observerUnsecurePort)),
+		ForcePull:  false,
+		Config:     containerConfig,
+		HostConfig: containerHostConfig,
+		Name:       conf.ObserverName,
 	}
 
-	suiteContainer := net.suite.Container(containerOpts)
-
 	nodeContainer := &Container{
 		Ports:   make(map[string]string),
 		datadir: tmpdir,
 		net:     net,
 		opts:    &containerOpts,
 	}
 
+	hostGRPCPort := nodeContainer.exposePort(t, GRPCPort, "9000/tcp")
+	nodeContainer.exposePort(t, GRPCSecurePort, "9001/tcp")
+	nodeContainer.exposePort(t, GRPCWebPort, "8000/tcp")
+
+	nodeContainer.opts.HealthCheck = testingdock.HealthCheckCustom(healthcheckAccessGRPC(hostGRPCPort))
+
+	suiteContainer := net.suite.Container(containerOpts)
 	nodeContainer.Container = suiteContainer
 	net.Containers[nodeContainer.Name()] = nodeContainer
 
-	net.network.After(suiteContainer)
+	// start after the bootstrap access node
+	accessNode.After(suiteContainer)
```

Comment:
I think this is what was causing the flakiness. Observers will fail to start their gRPC server if the proxied access node is not available.